src/GitTools.Core/GitTools.Core.Shared/Git/GitRepositoryFactory.cs

```diff
@@ -6,51 +6,62 @@ namespace GitTools.Git
     using LibGit2Sharp;
     using Logging;
 
-    public class GitPreparer
+    public static class GitRepositoryFactory
```

Comment:
Am just having a play with this now, I think this only works for dynamic repositories? A non-dynamic repo will never get normalised?

Also does noFetch make sense for dynamic repos?